constraints.txt

```diff
@@ -1,73 +0,0 @@
-# Specifies Python package versions for testing.
```

Comment:
One thing I just happened to notice is this file is referenced in `docs/requirements.txt` which is used by readthedocs. We should fix this somehow, but I'm not immediately sure the best way to do it. I think we should keep our dependencies pinned on readthedocs if at all possible.

Reply:
This was surprisingly easy. Thanks to the `.readthedocs.yaml` file that you can put at the root of the repo, you can tweak the doc build process made by ReadTheDoc.

So I used this mechanism to basically install poetry, export the constraints.txt file at the expected place for docs/requirements.txt, and given that pip supports now editable installs of PEP-517 compliant projects, ... voilà!